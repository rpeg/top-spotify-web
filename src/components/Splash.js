import React from 'react';

const ExampleChart = () => (
  <div className="splash">
    <div className="splash-header">
      <h1 className="splash-title">Spotify Data Visualizer</h1>
      <p className="splash-desc">
        Create a custom chart displaying your Spotify data, including top artists, tracks, genres, and stats.
      </p>
    </div>
    <div
      className="splash-example"
      style={{
        backgroundImage: `linear-gradient(to bottom, rgba(25, 20, 20, 0.3), rgba(25, 20, 20, 1)), 
          url(${process.env.PUBLIC_URL}images/example-chart.png`,
      }}
    />
  </div>
);

export default ExampleChart;
