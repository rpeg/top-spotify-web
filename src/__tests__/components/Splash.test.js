import React from 'react';
import { shallow } from 'enzyme';

import Splash from '../../components/Splash';

describe('Splash', () => {
  it('should not log errors in console', () => {
    const wrapper = shallow(<Splash />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Splash />);
    expect(wrapper).toMatchSnapshot();
  });
});
