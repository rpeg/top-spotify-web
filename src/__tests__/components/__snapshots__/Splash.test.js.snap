// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash renders correctly 1`] = `
<div
  className="splash"
>
  <div
    className="splash-header"
  >
    <p
      className="text-xl splash-title"
    >
      Spotify Data Visualizer
    </p>
    <p
      className="text-m splash-desc"
    >
      Create a custom chart displaying your Spotify data, including top artists, tracks, genres, and stats.
    </p>
  </div>
  <div
    className="splash-example"
    style={
      Object {
        "backgroundImage": "linear-gradient(to bottom, rgba(25, 20, 20, 0.8), rgba(25, 20, 20, 1)), 
          url(images/example-chart.png",
      }
    }
  />
</div>
`;
